### ISSUE TITLE
Show PR review status

### ISSUE THREAD

**Issue by rebornix** [microsoft/vscode-pull-request-github#681]
- Extension version: Insiders
- VSCode Version: 1.29
- OS: macOS

Now we are using Pull Requests for features and bug fixes always, so I still rely on emails and .com to know if my pr is approved and ready to be merged or not. If we can show indicators of the review status in PRs tree view, I can simply go to category *Created By Me*.

**Comment by StanleyGoldman** [comment]
@queerviolet and I were plotting on possible node structures we are partial to the first, but I thought we could put the others up here for discussion.

``` 
- [Checks Icon] [Review Icon] PR Title #xx by @StanleyGoldman
  - Desc
  - File 1
  - File 2
  - File N

- [Avatar] PR Title #xx by @StanleyGoldman
  - Desc
  - Reviewers
    - Ashi: +1
    - Stanley: -1
  - Checks
    - Check 1: +1
    - Check 2: +1
  - File 1
  - File 2
  - File N
  
- [Avatar] PR Title #xx by @StanleyGoldman
  - Desc
  - Reviewers: +1/-1/M
  - Checks:  +1/-1/M
  - File 1
  - File 2
  - File N
```

**Comment by StanleyGoldman** [comment]
Displaying two icons in the tree would not be ideal from a UX standpoint. We are trying to figure out how to merge both sets of states into one icon.

Checks states:
1. Passing
2. Failture
3. Pending

Review states:
1. Approve
2. Request Changes
3. Review Requested
4. Required

Both of those states are not all of what goes into "merge-ability".

**Comment by StanleyGoldman** [comment]
The simplest answer could very well be to display an icon for merge-ability directly.

**Comment by queerviolet** [comment]
I think there are really three basic states we want to represent:
  - ✅This PR is good to merge
  - ⛔️This PR can't be merged:
    - checks are failing, so fix them
    - a review is required, so assign reviewers
    - a reviewer requested changes, so make them
  - ⚠️Waiting on:
    - A review from a requested reviewer
    - Checks to complete

**Comment by StanleyGoldman** [comment]
Okay, so I've been looking deeper into the initial desire and comparing with other Pull Request centric GitHub plugins do.

Both plugins:
1. Keep the users avatar in a primary position
1. Display Pull Request Checks to the right instead of a Ready to Merge indicator.
  a. The right side margin is also maintained as the panels shrink horizontally

GitHub for Visual Studio:
![image](https://user-images.githubusercontent.com/417571/53881798-8fd88000-3fe2-11e9-98df-d7c5a7213666.png)
![image](https://user-images.githubusercontent.com/417571/53882198-a92dfc00-3fe3-11e9-923f-70b85e511bdb.png)

GitHub for Atom:
![image](https://user-images.githubusercontent.com/417571/53881823-a252b980-3fe2-11e9-8710-aeb62fedcb7d.png)
![image](https://user-images.githubusercontent.com/417571/53882234-c6fb6100-3fe3-11e9-9195-41363a592622.png)

Also Visual Studio code already displays an additional checkmark (in text) to the right of a user's avatar to indicate the given pull request is the current checked out branch.

![image](https://user-images.githubusercontent.com/417571/53882531-8d772580-3fe4-11e9-975f-7e631e968f1a.png)

**TLDR**:
- For consistency, I don't think we should remove the user's avatar in favor of a different status indicator.
- I would like to add a similar status indicator to the right side of the pull request tree node if possible.
- I wonder if we could benefit from adding additional informational nodes under the pull request.

**Comment by meaghanlewis** [comment]
👋 @misolori do you have any feedback on what @StanleyGoldman described above? Specifically about adding a status indicator to the right side of the pull request tree node.

**Comment by misolori** [comment]
I think I like having the indicator in a consistent place, however we currently use a ✔️ to indicate when a PR is checked out, so that would be confusing to have two (we'd have to find a different way to differentiate that)

**Comment by srescio** [comment]
> I think I like having the indicator in a consistent place, however we currently use a ✔️ to indicate when a PR is checked out, so that would be confusing to have two (we'd have to find a different way to differentiate that)

could maybe change the "checked out" indicator to **bolded PR title**? it would even stand out more, I think sourceTree app does that to tell you which branch you're on

**Comment by mix3d** [comment]
Why not use a branch style icon for the checked out branch, on top of the bolded title? There's nothing stopping someone from putting a ✔️ in the commit title, which would be confusing anyways.

**Comment by mix3d** [comment]
A quick mockup mixing many of the similar suggestions. 
![image](https://user-images.githubusercontent.com/1474137/79790274-59ea7e00-8300-11ea-9335-9f608b33401c.png)
Notably missing is the PR creation date, which can be helpful when sorting through "which of these needs my attention most", though the PR # should be sufficient. 

Noting that when opening one of the PRs, the immediate next row is prepended with a similar branch style icon top open the in-vscode PR editor, I would suggest either replacing with a github logo (similar to the plugin's logo), or remove the icon altogether, leaving "description" in bold and actually pulling in the first bit of the PR message body. 

Considering most people don't have dozens of PRs assigned to them at a given time, I'm wondering if the two-row layout is more useful; what more information could you add there? (like the Visual Studio example).

**Comment by digitarald** [comment]
This should also consider _in-progress_ checks state, to make it easier to follow CI/CD results for your own PRs and the ones you are asked to review.

<img width="144" alt="image" src="https://user-images.githubusercontent.com/8599/193339593-1657f35d-ecc6-499d-9cb7-e632b4348d1c.png">

<img width="500" alt="image" src="https://user-images.githubusercontent.com/8599/193339483-bcbc82c9-ce3f-4d83-b098-8b9b26ec2666.png">

**Comment by alexr00** [comment]
We're limited to unicode characters for the status, here's what I have so far. Circle is pending, shaded square is review required, X is CI failed.

![image](https://user-images.githubusercontent.com/38270282/201700304-03494def-3d92-4300-95e0-26badd47c644.png)

**Comment by alexr00** [comment]
With codicons and colors instead:

![image](https://user-images.githubusercontent.com/38270282/201912718-4f2ad305-e4a4-4540-9eae-0d7b1917ce58.png)

**Comment by alexr00** [comment]
Tweaking some icons and the color of the checks icon:

![image](https://user-images.githubusercontent.com/38270282/202484171-396d8d4e-ef0c-4fb0-abd9-77532698698e.png)

@daviddossett, what do you think of the `git-pull-request-new-changes` icon for "changes requested"? Alternatively, should we have a version of the icon without the dot on it?

**Comment by mix3d** [comment]
What a journey! Love it. Thanks for all the work.

**Comment by daviddossett** [comment]
> @daviddossett, what do you think of the git-pull-request-new-changes icon for "changes requested"? Alternatively, should we have a version of the icon without the dot on it?

I see we have these as of today:
<img width="361" alt="CleanShot 2022-11-21 at 15 51 18@2x" src="https://user-images.githubusercontent.com/25163139/203181162-dfaf9a1a-f598-4582-9aac-47806b44536b.png">

..should we use `request-changes`? Or should that icon be updated to the "file" style pictured above if not used elsewhere?

**Comment by alexr00** [comment]
I missed the `request-changes` icon, that seems perfect. I'll update to use it.
